doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,66 +1,165 @@
+the colon, as for this theorem from Coq's standard library:
+
+.. coqtop:: all
+
+   Print proj1.
 
 .. _proof-editing-mode:
 
-Entering and leaving proof editing mode
----------------------------------------
+Entering and exiting proof mode
+-------------------------------
+
+Coq enters *proof mode* when you begin a proof through
+commands such as :cmd:`Theorem` or :cmd:`Goal`.  Coq user interfaces
+usually have a way to indicate that you're in proof mode.
+For example, CoqIDE shows the name of the theorem being proved at the
+bottom left of the window.  Coqtop changes its input prompt to
+include the name of the theorem being proven.
+
+.. todo PR: the last 3 sentences above seem pretty unimportant
+   don't you think the user will figure this out on their own?
+
+Tactics are only available only in proof mode, while commands
+are always available.  By convention, tactics begin with lowercase letters
+while commands begin with uppercase letters.
+
+.. todo PR: is the following really true?  I see that Focus is accepted outside a proof:
+
+   In Coq’s proof mode ... user has access to specialized
+   commands dealing with proof development pragmas documented in this
+   section.
 
-The proof editing mode is entered by asserting a statement, which typically is
-the assertion of a theorem using an assertion command like :cmd:`Theorem`. The
-list of assertion commands is given in :ref:`Assertions`. The command
-:cmd:`Goal` can also be used.
+.. todo PR: following seems to conflate the proof context with section-level info, e.g.
+   Parameter, ... are part of the section  Is this truly the "local context"?
+
+   To each goal is associated a number of hypotheses called the *local context*
+   of the goal. Initially, the local context contains the local variables and
+   hypotheses of the current section (see Section :ref:`gallina-assumptions`) and
+   the local variables and hypotheses of the theorem statement.
+
+When the proof is completed, you can exit proof mode with commands such as
+:cmd:`Qed`, :cmd:`Defined` and :cmd:`Save`.
 
 .. cmd:: Goal @type
 
-   This is intended for quick assertion of statements, without knowing in
-   advance which name to give to the assertion, typically for quick
-   testing of the provability of a statement. If the proof of the
-   statement is eventually completed and validated, the statement is then
-   bound to the name ``Unnamed_thm`` (or a variant of this name not already
-   used for another statement).
+   Asserts an unnamed proposition.  This is intended for quick tests that
+   a proposition is provable.  If the proof is eventually completed and
+   validated, you can assign a name with the :cmd:`Save` or :cmd:`Defined`.
+   commands.  If no name is given, the name will be `Unnamed_thm` (or,
+   if that name is already defined, a variant of that).
 
 .. cmd:: Qed
 
-   This command is available in interactive editing proof mode when the
+   This command is available in proof mode when the
```

Comment:
What is lacking from this description and from the introduction is that calling `Qed` will trigger the kernel of Coq, which will verify that the proof term is of the expected type.